Prioritize OIDC mechanism when inclusive authentication is disabled to simplify using mTLS and OIDC together (#45072)
quarkusio/quarkus: docs/src/main/asciidoc/security-authentication-mechanisms.adoc

If the authentication is inclusive then `SecurityIdentity` created by the first authentication mechanism can be
injected into the application code.
For example, if both <<mutual-tls>> and basic authentication mechanism authentications are required,
the <<mutual-tls>> mechanism will create `SecurityIdentity` first.

> NOTE: The <<mutual-tls>> mechanism has the highest priority when inclusive authentication is enabled, so that other

Reviewer: ```suggestion
NOTE: The <<mutual-tls>> mechanism has the highest priority when inclusive authentication is enabled, to ensure
```

If the authentication is inclusive then `SecurityIdentity` created by the first authentication mechanism can be
injected into the application code.
For example, if both <<mutual-tls>> and basic authentication mechanism authentications are required,
the <<mutual-tls>> mechanism will create `SecurityIdentity` first.

NOTE: The <<mutual-tls>> mechanism has the highest priority when inclusive authentication is enabled, to ensure
that an injected `SecurityIdentity` always represents <<mutual-tls>> and can be used to get access to `SecurityIdentity`

> identities provided by other authentication mechanisms with a `io.quarkus.vertx.http.runtime.security.HttpSecurityUtils.getSecurityIdentities(SecurityIdentity identity)` call.

Author: No, I was thinking the same about that being already explained, but I just wanted to apply your changes and move on :-). Fixed.
